src/components/utils/src/threads/thread_posix.cc

```diff
@@ -275,10 +278,13 @@ void Thread::join() {
 }
 
 Thread::~Thread() {
-  finalized_ = true;
-  stopped_ = true;
-  join();
-  pthread_join(handle_, NULL);
+  if (!finalized_) {
```

Comment:
@orhan-mehmedov , finalized_ will be changed inside if, it garanties that pthread_join() will be called only ones.